bench/fixture.ts

```diff
@@ -57,10 +68,46 @@ export class TokenManager {
     await token.mint(settlement.address, LOTS);
 
     for (const trader of this.traders) {
-      await token.mint(trader.address, LOTS);
+      await token.mint(trader.address, LOTS.mul(3));
       await token.connect(trader).approve(vaultRelayer.address, LOTS);
+      await token.connect(trader).approve(vault.address, LOTS.mul(2));
+      await vault.connect(trader).depositToInternalBalance([
+        {
+          token: token.address,
+          amount: LOTS,
+          sender: trader.address,
+          recipient: trader.address,
+        },
+      ]);
     }
 
+    await balancerPool.registerTokens(
+      [token.address],
+      [ethers.constants.AddressZero],
+    );
+    await token.mint(pooler.address, LOTS);
+    await token.connect(pooler).approve(vault.address, LOTS);
+    const existingTokens = this.instances.map(({ address }) => address);
+    const zeros = this.instances.map(() => 0);
+    await vault.connect(pooler).joinPool(
+      await balancerPool.getPoolId(),
+      pooler.address,
+      pooler.address,
+      [...existingTokens, token.address],
```

Comment:
Why do we need to provide all previous tokens here?

Reply:
When joining the pool, all tokens must be specified.